sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveLateralColumnAliasReference.scala

```diff
@@ -168,19 +168,18 @@ object ResolveLateralColumnAliasReference extends Rule[LogicalPlan] {
             && aggregateExpressions.exists(_.containsPattern(LATERAL_COLUMN_ALIAS_REFERENCE)) =>
 
           // Check if current Aggregate is eligible to lift up with Project: the aggregate
-          // expression only contains: 1) aggregate functions, 2) grouping expressions, 3) lateral
-          // column alias reference or 4) literals.
+          // expression only contains: 1) aggregate functions, 2) grouping expressions, 3) leaf
+          // expressions excluding attributes not in grouping expressions
           // This check is to prevent unnecessary transformation on invalid plan, to guarantee it
           // throws the same exception. For example, cases like non-aggregate expressions not
           // in group by, once transformed, will throw a different exception: missing input.
           def eligibleToLiftUp(exp: Expression): Boolean = {
             exp match {
               case e if AggregateExpression.isAggregate(e) => true
-              case e if groupingExpressions.exists(_.semanticEquals(e)) => true
-              case _: Literal | _: LateralColumnAliasReference => true
+              case e: Attribute if !groupingExpressions.exists(_.semanticEquals(e)) => false
```

Comment:
can we move it closer to `case e if groupingExpressions.exists(_.semanticEquals(e)) => true`?

Reply:
and do we really need both cases? I think
```
case e if groupingExpressions.exists(_.semanticEquals(e)) => true
case a: Attribute => false
```
should be sufficient?

Reply:
I follow (copy) the condition in checkAnalysis, but yes I think yours is sufficient. Will change.